### PR TITLE
add new feed for relock alarm

### PR DESCRIPTION
**Pull request by yoshinori-0778**
This PR is to add new feed for relock.

## Description

Based on the feeds (number of good/all detectors) we have, just add new feed being calculated as 1 if bad(all - good) > good, else 0.

## Motivation and Context

This is for new alarm that check relock quality.
In terms of making new alarm of this, calculating this with the feeds (number of good/all detectors) we have after publishing is computational intense. Thus, I would like to make those values here and publish them in addition to the current feeds

## How Has This Been Tested?

Not yet on the site computer but I hope we do not need to test because this is simple change.
Script itself has been tested as below, which is expected.

```
alls = np.array([100,100,100,100,100,100])
goods = np.array([30,40,49,50,51,80])
block_data = {}
for iband, (iall, igood) in enumerate(zip(alls, goods)):
    block_data[f'alldet_band{iband}'] = iall
    block_data[f'gooddet_band{iband}'] = igood
    block_data[f'alarm_band{iband}'] = 1 if (iall - igood) > igood else 0
    #print(iall - igood, igood)

block_data -->
{'alldet_band0': 100,
 'gooddet_band0': 30,
 'alarm_band0': 1,
 'alldet_band1': 100,
 'gooddet_band1': 40,
 'alarm_band1': 1,
 'alldet_band2': 100,
 'gooddet_band2': 49,
 'alarm_band2': 1,
 'alldet_band3': 100,
 'gooddet_band3': 50,
 'alarm_band3': 0,
 'alldet_band4': 100,
 'gooddet_band4': 51,
 'alarm_band4': 0,
 'alldet_band5': 100,
 'gooddet_band5': 80,
 'alarm_band5': 0}
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by yoshinori-0778** [comment]
Thanks for checking, Brian.
This alarm would be one alarm for each slot, and each alarm is checking all 8 bands for that slot and would fire if any of the 8 bands pass the threshold, so 7 alarms for one SAT. @davidvng can comment as well.

50/50 threshold is based on our experience. If Relock looks bad, it was completely bad (i.e., number of bad dets >> number of good dets), so this threshold should be good to detect it as a first step. Of course we can change it later as you commentd.

**Comment by BrianJKoopman** [comment]
> Do you mean I should add number of 50% (e.g. `int(f'alldet_band{iband}/2'` )) instead of adding `alarm_band{iband}` that I added?

No, more like:
```
block_data[f'alarm_band{iband}'] = 1 if igood/iall < good_threshold else 0
```

where `good_threshold` gets passed in via the config file, so we have some control over how sensitive the alarm is -- though it sounds like it's either they're mostly bad or they are mostly good.

**Comment by yoshinori-0778** [comment]
I see, that makes sense

Where is the config file? https://github.com/simonsobs/sodetlib/blob/master/sodetlib/det_config.py? or something new?
Once I can get it, I will modify the script and config file.

**Comment by BrianJKoopman** [comment]
> Where is the config file? https://github.com/simonsobs/sodetlib/blob/master/sodetlib/det_config.py? or something new? Once I can get it, I will modify the script and config file.

I was originally thinking in the ocs site config file (SCF), but that is an awkward mechanism for single task items like this, since it requires making the option a command line argument.

I'm not against putting it in the `sys_config.yml` as you suggest, that sounds like it would be simplest. I would say try to load it from there and otherwise default to 50%. That way we don't have to go update all the sys_config files unless we need to change this initial value.

**Comment by BrianJKoopman** [comment]
On second thought, I'm not sure it really belongs there either, that would require updates to the [sodetlib documentation for the structure of that configuration file](https://sodetlib.readthedocs.io/en/latest/configs.html#docker-environment).

Really what we need is a more general config file format for ocs agents, but that isn't going to be ready for this application.

For now, the best thing would be to put it in the SCF, even if it is awkward. That involves adding an argument to the parser here: https://github.com/simonsobs/socs/blob/f40b3355d807b1094ef64dadc402cbd864282e5d/socs/agents/pysmurf_controller/agent.py#L1208-L1223

The whole arguments list gets passed to the `PysmurfController` object when it's created, you can then use it from there.

**Comment by yoshinori-0778** [comment]
Thank you for your comments, Brian!
I modified the script based on you comments.
Is this what you expected?

File: socs/agents/pysmurf_controller/agent.py [hunk 1]
```diff
@@ -1220,6 +1223,8 @@ def make_parser(parser=None):
                         help="Smurf slot that this agent will be controlling")
     pgroup.add_argument('--poll-interval', type=float,
                         help="Time between check-state polls")
+    pgroup.add_argument('--good_threshold', type=float, default=0.5,
+                        help="Time between check-state polls")
```
Comment (BrianJKoopman) [inline socs/agents/pysmurf_controller/agent.py:1227]: Update the 'help' here to describe the new argument.

Comment (yoshinori-0778) [reply socs/agents/pysmurf_controller/agent.py]: Thank you for catching it, fixed.

**Comment by yoshinori-0778** [comment]
Thanks, Brian!
Do you think we need to test on site computers before merge?
If not I would like to merger it into main once current checks are finished.

**Comment by BrianJKoopman** [comment]
> Thanks, Brian! Do you think we need to test on site computers before merge? If not I would like to merger it into main once current checks are finished.

I think we're good, this is a small change and you already caught a good edge case from the tests. I'm planning on merging once the checks complete.

**Comment by yoshinori-0778** [comment]
Sounds great, thanks Brian!

**Comment by yoshinori-0778** [comment]
Once this is implemented on the site computer, could you let me know so I can ask David to make the alarm.
Thanks!
